src/Microsoft.AspNet.Authentication.Cookies/CookieAuthenticationHandler.cs

```diff
@@ -299,7 +314,7 @@ private CookieOptions BuildCookieOptions()
             catch (Exception exception)
             {
                 var exceptionContext = new CookieExceptionContext(Context, Options,
-                    CookieExceptionContext.ExceptionLocation.ApplyResponseGrant, exception, model);
+                    CookieExceptionContext.ExceptionLocation.ApplyResponseGrant, exception, ticket);
```

Comment:
Added enums for each virtual (SignIn, SignOut, Unauthorized, Forbidden, FinishResponse) this seems a bit much to me...